lib/request/complement.js

```diff
@@ -25,9 +23,9 @@ module.exports = function complement(requested, optimized, tree) {
         var subTree = tree[oPath.length];
 
         var intersectionData = findPartialIntersections(rPath, oPath, subTree);
-        intersection = arrayConcat(intersection, intersectionData[0]);
-        optimizedComplement = arrayConcat(optimizedComplement, intersectionData[1]);
-        requestedComplement = arrayConcat(requestedComplement, intersectionData[2]);
```

Comment:
Since these arrays are local, and we're reassigning them to the return value of concat, we can just mutate the original arrays instead of creating new ones.